src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,149 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.RegistrationException;
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static final String LARGE_AGE_MESSAGE = "Incorrect age!";
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String SHORT_LOGIN = "short";
+    private static final String SHORT_PASSWORD = "short";
+    private static final int VALID_AGE = 23;
+    private static final int UNDERAGE = 16;
+    private static final int NEGATIVE_AGE = -20;
+    private static final int ZERO_AGE = 0;
+    private static final int OVER_AGE = 150;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void loginNullValue_NotOk() {
+        User user = createNewUser(null, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void existingLogin_NotOk() {
+        User user = createNewUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        storageDao.add(user);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerShortLogin_NotOk() {
+        User user = createNewUser(SHORT_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
```

Comment:
The test `registerShortLogin_NotOk` does not cover an edge case for the login length. According to the checklist, it's important to test edge cases like minimum length requirements.